### ISSUE TITLE
Configuration will be overwritten

### ISSUE THREAD

**Issue by JHCD** [flobz/psa_car_controller#243]
**Describe the bug**
This is a follow-up ticket for discussion of the loading issues with the docker image:
When the Docker container is restarted, the configuration is overwritten.

I've done a few more tests and have no idea what might be the cause.
I think the problem lies elsewhere than originally thought.

I run dozens of Docker containers on different QNAP servers and can reproduce this behavior on all of my QNAPs.

**Test procedure:**
1.  Docker container installed
2.  Config and OTP set up
3.  Docker terminated
4. API keys added
5. Docker started

As you can see from the console, the config-path is correctly mapped into the container and the content of the config file including all API keys is displayed with cat.

The following error is now output: `ERROR :: Please reconnect by going to config web page`
A check of the file shows that the config file has not yet been touched.

As soon as I visit the website, I am redirected to the config page and have to insert my credentials. 
After I pressed Submit, the OTP data is requested and the configuration file is overwritten.

It looks as if all data from the old configuration will be replaced by the default values and not the old values will be reused.

**init.sh for debugging:**
```bash
#!/bin/bash
echo " "
echo "--------- NEW -----------"
echo " "
echo "Envs: "
env

echo "Containerised psa_car_controller loading..."

cd "$PSACC_CONFIG_DIR"
ARGS="-p $PSACC_PORT -l 0.0.0.0 -b $PSACC_BASE_PATH $PSACC_OPTIONS"
echo "Paramter: "
echo $ARGS

echo "Show Dir and Config:"
ls -al /config
cat /config/config.json

echo "Start Server..."

python3 -u /psa_car_controller/server.py $ARGS
```

**Output Console start docker**
```
--------- NEW -----------
 
Envs: 
HOSTNAME=psa_car_controller-1
PWD=/config
PSACC_CONFIG_DIR=/config
PSACC_OPTIONS=-c -r --web-conf
HOME=/root
PSACC_BASE_PATH=/
TERM=xterm
SHLVL=1
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PSACC_PORT=5000
_=/usr/bin/env

Containerised psa_car_controller loading...

Paramter: 
-p 5000 -l 0.0.0.0 -b / -d -c -r --web-conf

Show Dir and Config:
total 71752
drwxrwxrwx 4 root root     4096 Oct  9 14:28 .
drwxr-xr-x 1 root root     4096 Oct  9 14:04 ..
-rw-rw-rw- 1 root root     1026 Oct  9 14:23 cars.json
drwxr-xr-x 2 root root     4096 Oct  9 14:05 certs
-rw-rw-rw- 1 root root       99 Oct  7 11:15 charge_config.json
-rw-rw-rw- 1 root root      114 Oct  3 11:14 config.ini
-rw-rw-rw- 1 root root      688 Oct  9 14:23 config.json
-rw-rw-rw- 1 root root    40960 Oct  9 14:23 info.db
-rw-rw-rw- 1 root root 73393862 Oct  3 11:14 myopel.apk
-rw-rw-rw- 1 root root     3075 Oct  7 09:47 otp.bin

{
    "abrp": {
        "abrp_enable_vin": [
            "VXKUHZKXZM..."
        ],
        "token": "[xxx]"
    },
    "client_id": "[xxx]",
    "client_secret": "[xxx]",
    "co2_signal_api": "[xxx]",
    "country_code": "DE",
    "customer_id": "[xxx]",
    "proxies": {
        "http": "",
        "https": ""
    },
    "realm": "clientsB2COpel",
    "refresh_token": "[xxx]",
    "remote_refresh_token": "[xxx]",
    "weather_api": "[xxx]"
}

Start Server...
/usr/local/lib/python3.7/dist-packages/dash_bootstrap_components/_table.py:5: UserWarning: 
The dash_html_components package is deprecated. Please replace
`import dash_html_components as html` with `from dash import html`
  import dash_html_components as html
2021-10-09 14:31:49,223 :: WARNING :: Can't get car model, please report an issue on github with your car model and first ten letter of your VIN : VXEVEEHZ8L
2021-10-09 14:31:49,576 :: ERROR :: Please reconnect by going to config web page
2021-10-09 14:31:49,578 :: INFO :: save config change
2021-10-09 14:31:49,585 :: WARNING :: Can't get language
WARNING: dev bundles requested with serve_locally=False.
This is not supported, switching to serve_locally=True
/psa_car_controller/web/view/views.py:8: UserWarning: 
The dash_core_components package is deprecated. Please replace
`import dash_core_components as dcc` with `from dash import dcc`
  import dash_core_components as dcc
/psa_car_controller/web/figures.py:4: UserWarning: 
The dash_table package is deprecated. Please replace
`import dash_table` with `from dash import dash_table`

Also, if you're using any of the table format helpers (e.g. Group), replace 
`from dash_table.Format import Group` with 
`from dash.dash_table.Format import Group`
  import dash_table
2021-10-09 14:31:51,500 :: INFO :: update_data
 * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://x.x.x.x:5000/ (Press CTRL+C to quit)
```

**Output Console open webpage and edit config**
```
x.x.x.x - - [09/Oct/2021 14:45:51] "GET / HTTP/1.1" 200 -
x.x.x.x - - [09/Oct/2021 14:45:51] "GET /assets/clientside.js?m=1632161517.0 HTTP/1.1" 200 -
x.x.x.x - - [09/Oct/2021 14:45:52] "GET /_dash-dependencies HTTP/1.1" 200 -
x.x.x.x - - [09/Oct/2021 14:45:52] "GET /_dash-layout HTTP/1.1" 200 -
x.x.x.x - - [09/Oct/2021 14:45:52] "GET /_favicon.ico?v=2.0.0 HTTP/1.1" 200 -
x.x.x.x - - [09/Oct/2021 14:45:52] "POST /_dash-update-component HTTP/1.1" 200 -
x.x.x.x - - [09/Oct/2021 14:45:52] "GET /config HTTP/1.1" 200 -
x.x.x.x - - [09/Oct/2021 14:45:52] "GET /assets/clientside.js?m=1632161517.0 HTTP/1.1" 304 -
x.x.x.x - - [09/Oct/2021 14:45:52] "GET /_dash-layout HTTP/1.1" 200 -
x.x.x.x - - [09/Oct/2021 14:45:52] "GET /_dash-dependencies HTTP/1.1" 200 -
x.x.x.x - - [09/Oct/2021 14:45:52] "POST /_dash-update-component HTTP/1.1" 200 -
x.x.x.x - - [09/Oct/2021 14:45:52] "POST /_dash-update-component HTTP/1.1" 200 -
2021-10-09 14:46:35,964 :: WARNING :: Can't get car model, please report an issue on github with your car model and first ten letter of your VIN : VXEVEEHZ8L
2021-10-09 14:46:36,361 :: INFO :: save config change
2021-10-09 14:46:36,799 :: WARNING :: Can't get car model, please report an issue on github with your car model and first ten letter of your VIN : VXEVEEHZ8L
2021-10-09 14:46:36,801 :: INFO :: 
Your vehicles: ["{'vin': 'VXKUHZKXZM', 'vehicle_id': '', 'label': 'corsa-e', 'brand': 'C', 'abrp_name': 'opel:corsae:20:50', 'battery_power': 46, 'fuel_capacity': 0, 'max_elec_consumption': 70, 'max_fuel_consumption': 0}", "{'vin': 'VXEVEEHZ8L', 'vehicle_id': '', 'label': 'Life', 'brand': 'C', 'abrp_name': None, 'battery_power': 46, 'fuel_capacity': 0, 'max_elec_consumption': 70, 'max_fuel_consumption': 30}"]
2021-10-09 14:46:36,803 :: INFO :: save config change
2021-10-09 14:46:37,086 :: WARNING :: Can't get car model, please report an issue on github with your car model and first ten letter of your VIN : VXEVEEHZ8L
2021-10-09 14:46:37,086 :: WARNING :: Can't get car model, please report an issue on github with your car model and first ten letter of your VIN : VXEVEEHZ8L
2021-10-09 14:46:37,403 :: INFO :: save config change
2021-10-09 14:46:37,403 :: INFO :: save config change
2021-10-09 14:46:37,918 :: WARNING :: Can't get car model, please report an issue on github with your car model and first ten letter of your VIN : VXEVEEHZ8L
2021-10-09 14:46:37,918 :: WARNING :: Can't get car model, please report an issue on github with your car model and first ten letter of your VIN : VXEVEEHZ8L
2021-10-09 14:46:37,920 :: INFO :: ["{'vin': 'VXKUHZKXZM', 'vehicle_id': '', 'label': 'corsa-e', 'brand': 'C', 'abrp_name': 'opel:corsae:20:50', 'battery_power': 46, 'fuel_capacity': 0, 'max_elec_consumption': 70, 'max_fuel_consumption': 0}", "{'vin': 'VXEVEEHZ8L', 'vehicle_id': '', 'label': 'unknown', 'brand': 'C', 'abrp_name': None, 'battery_power': 46, 'fuel_capacity': 0, 'max_elec_consumption': 70, 'max_fuel_consumption': 30}"]
2021-10-09 14:46:37,920 :: INFO :: ["{'vin': 'VXKUHZKXZM', 'vehicle_id': '', 'label': 'corsa-e', 'brand': 'C', 'abrp_name': 'opel:corsae:20:50', 'battery_power': 46, 'fuel_capacity': 0, 'max_elec_consumption': 70, 'max_fuel_consumption': 0}", "{'vin': 'VXEVEEHZ8L', 'vehicle_id': '', 'label': 'unknown', 'brand': 'C', 'abrp_name': None, 'battery_power': 46, 'fuel_capacity': 0, 'max_elec_consumption': 70, 'max_fuel_consumption': 30}"]
2021-10-09 14:46:38,309 :: INFO :: save config change
2021-10-09 14:46:38,309 :: INFO :: save config change
2021-10-09 14:46:38,311 :: ERROR :: remote_refresh_token isn't defined
2021-10-09 14:46:38,311 :: ERROR :: remote_refresh_token isn't defined
2021-10-09 14:46:38,990 :: ERROR :: Error during activation: {'newversion': '2.0.0', 'newversionurl': 'http://m.inwebo.com/', 'err': 'NOK:ACCESS'}
2021-10-09 14:46:38,990 :: ERROR :: Error during activation: {'newversion': '2.0.0', 'newversionurl': 'http://m.inwebo.com/', 'err': 'NOK:ACCESS'}
What is the code you just received by SMS ?
```

**Comment by flobz** [comment]
I close the issue due to inactivity.

**Comment by JHCD** [comment]
Did you solve it?

**Comment by flobz** [comment]
No. This problem appear only on Qnap and I don't have one. "I close the issue due to inactivity."

**Comment by JHCD** [comment]
Since the data is correctly available in Docker before the UI is opened, I consider the statement to be incorrect.

**Comment by flobz** [comment]
You are the only one with this problem apparently and you don't fix it so I close the issue due to inactivity.
I can't do more.

**Comment by JHCD** [comment]
How can I check which files your code is reading? Can I see what the server is reading in? 
I can't do more than look at the files in parallel in the Linux console. They are correct until I open the UI.

My problem is that I don't want to make endless OTP requests. How can I save / reload the old request if the server has thrown the settings away? I don't want to deactivate the PSA account because of the 20 SMS activations.
